### PR TITLE
Map FERC 714 XBRL and CSV IDs

### LINKED ISSUE #3846: Synchronize FERC 714 XBRL and DBF IDs and combine into one table

### PR DESCRIPTION
**Pull request by e-belfer**
# Overview

Closes #3846.

What problem does this address?
Creates a glue CSV mapping the IDs of FERC 714 respondents in XBRL and CSV data. This builds on the migration mapping noted in #3846.

The [migrated data](https://www.ferc.gov/filing-forms/eforms-refresh/migrated-data-downloads) page from FERC notes the following:
> Companies that did not have a CID prior to the migration have been assigned a CID that begins with R, i.e., a temporary RID.  These RIDs will be replaced in future with the accurate CIDs and new datasets will be published.

However, some of the IDs beginning with C in the migrated data weren't found in the actual XBRL data, while respondents matching the names and locations were found with different respondent IDs. Thus, I manually reviewed the IDs for each respondent, matching based on name. Some quirks to note:
- Unmatched respondents mostly occur due to mergers, splits, acquisitions, and companies that no longer exist.
- All respondents are matched 1:1 from CSV to XBRL data. For instance, Western Area Power Admin - Upper Missouri-East and Western Area Power Administration - Upper Missouri West in the CSV data merge into Western Area Power Administration - Upper Great Plains Region in the XBRL data, but these records are not linked.
- Where the migration data assigned an ID beginning with C that doesn't appear in the data, and no updated ID was found, I have left the ID in place in case it reappears in the XBRL data. All temporary IDs (beginning with R) were removed.

What did you change?
- Added a new CSV to the `pudl.package_data.glue` module.

# Testing

How did you make sure this worked? How can a reviewer verify this?

```[tasklist]
# To-do list
- [x] Review the PR yourself and call out any questions or issues you have
```

### CODE REVIEW

**Comment by e-belfer** [comment]
@aesharpe Think I should probably update docs somewhere along the way about the matching process, but let me know if you have thoughts about where in the docs a cleaned up version of these notes should live?

**Review by cmgosnell — APPROVED** [review]
🎉

**Comment by aesharpe** [comment]
> @aesharpe Think I should probably update docs somewhere along the way about the matching process, but let me know if you have thoughts about where in the docs a cleaned up version of these notes should live?

I think this should live in the "Notable Irregularities" section of the Data Source page for now. We don't have one for 714 yet, but either @cmgosnell or I was planning on adding one. I might make a separate PR branch for this that either of you can add to.

**Comment by e-belfer** [comment]
> > @aesharpe Think I should probably update docs somewhere along the way about the matching process, but let me know if you have thoughts about where in the docs a cleaned up version of these notes should live?
> 
> I think this should live in the "Notable Irregularities" section of the Data Source page for now. We don't have one for 714 yet, but either @cmgosnell or I was planning on adding one. I might make a separate PR branch for this that either of you can add to.

Great, that makes sense to me! As this is documented in the PR description I'll go ahead and merge this, and we can pull from this to add to the PR.

File: src/pudl/package_data/glue/respondent_id_ferc714.csv [hunk 1]
```diff
@@ -0,0 +1,218 @@
+respondent_id_ferc714,respondent_id_ferc714_xbrl,respondent_id_714_csv,Source,Notes
```
Comment (zaneselvans) [inline src/pudl/package_data/glue/respondent_id_ferc714.csv:1]: Is there a reason that `respondent_id_714_csv` doesn't follow the same naming convention as the other two ID columns?

Comment (e-belfer) [reply src/pudl/package_data/glue/respondent_id_ferc714.csv:1]: Sheer oversight - sounds like @cmgosnell already fixed this in her branch though.

**Comment by aesharpe** [comment]
@e-belfer you can add any future 714 docs changes to this PR https://github.com/catalyst-cooperative/pudl/pull/3850

**Pull request merged**